app/pages/amigo-requests/amigo-requests.ts

```diff
@@ -23,21 +23,55 @@ export class AmigoRequestsPage implements OnInit {
 
     public amigoShareRequests: Array<AmigoShareRequest>;
 
+    public accept: (AmigoShareRequest) => void;
+
+    public decline: (AmigoShareRequest) => void;
+
     constructor(
         private amigoShareRequestService: AmigoShareRequestService
     ) {
+        this.accept = this.acceptAmigoShareRequest.bind(this);
+        this.decline = this.declineAmigoShareRequest.bind(this);
```

Comment:
`this` burned me when I passed my callback functions into the amigo request cards.